core/domain/state_domain.py

```diff
@@ -2281,6 +2334,150 @@ def create_default_subtitled_unicode(cls, content_id):
         return cls(content_id, '')
 
 
+class SubtitledSetOfUnicodeString(python_utils.OBJECT):
```

Comment:
Not sure this is needed if primitive types are used (e.g. list of str).

Also, uniqueness should probably be checked in the relevant validate() method.